test/models/user/privilege_test.rb

```diff
@@ -0,0 +1,31 @@
+require "test_helper"
+
+class User::PrivilegeTest < ActiveSupport::TestCase
+  test "promoting a user" do
+    user = users(:peasant_matt)
+    user.promote_to_admin
+
+    assert user.admin?
+  end
+
+  test "promoting a user creates an event" do
+    user = users(:peasant_matt)
+    user.promote_to_admin
+
+    assert user.events&.first&.action == "promoted_to_admin"
```

Comment:
would it be best to go for the last event here?

Reply:
The database is reset after every test, but yeah, now that I think about it, that makes the concept clearer.